CDTDatastore/query/CDTQProjectedDocumentRevision.h

```diff
@@ -28,12 +30,14 @@
 /**
  Initialise with a datastore so mutableCopy can return a full document.
  */
-- (id)initWithDocId:(NSString *)docId
-         revisionId:(NSString *)revId
-               body:(NSDictionary *)body
-            deleted:(BOOL)deleted
-        attachments:(NSDictionary *)attachments
-           sequence:(SequenceNumber)sequence
-          datastore:(CDTDatastore *)datastore;
+- (nullable instancetype)initWithDocId:(NSString *)docId
```

Comment:
I don't believe `NSObject init` can return `nil`. Although it says in the [doc](https://developer.apple.com/library/mac/documentation/Cocoa/Reference/Foundation/Classes/NSObject_Class/#//apple_ref/occ/instm/NSObject/init):

> An initialized object, or nil if an object could not be created for some reason that would not result in an exception.

It goes on to say:

> The init method defined in the NSObject class does no initialization; it simply returns self.

So I think the former comment refers to `init` methods in general, whereas the latter refers to `NSObject init` specifically.